src/Features/CSharp/Portable/CodeRefactorings/InlineTemporary/InlineTemporaryCodeRefactoringProvider.cs

```diff
@@ -65,7 +65,8 @@ public override async Task ComputeRefactoringsAsync(CodeRefactoringContext conte
 
             if (variableDeclarator.Initializer == null ||
                 variableDeclarator.Initializer.Value.IsMissing ||
-                variableDeclarator.Initializer.Value.IsKind(SyntaxKind.StackAllocArrayCreationExpression))
+                variableDeclarator.Initializer.Value.IsKind(SyntaxKind.StackAllocArrayCreationExpression) ||
+                variableDeclarator.Initializer.Value.DescendantNodesAndSelf().Any(n => n.IsKind(SyntaxKind.ImplicitObjectCreationExpression)))
```

Comment:
this is not the right thing to do here.

First, this will block *any* usage of an implicit object creation.  for example, if you had `Foo f = new Foo(new());`.  Which is certainly not ok.

Second, even if you have `Foo f = new();` it's not necessarily the case that it's bad to allow inlining of this.  For example, if you have `Foo f = new(); Bar(f);`, then it may be fine to inline.

What we need to do instead is examine the points where the variable is inlined to and see if it would cause a problem.  if so, then we need to add an 'error annotation' there so that the preview calls out the problem.